src/app/services/transaction/transaction-base/transaction-base.component.ts

```diff
@@ -13,29 +13,34 @@ export class TransactionBaseComponent {
     public deleteDisabled: boolean;
     public editDisabled: boolean;
 
-    // inject the transaction service
+    public applyDelete: boolean;
+    public applyEdit: boolean;
```

Comment:
TBH, the checks are arbitrary for the demo, I'd be okay if they are visibly so:
```html
<igx-icon igxListAction (click)="onEdit()" *ngIf="item.id === 1 && item.price !== '$999'">edit</igx-icon>
```
And just drop the entire state variable thing that's sprinkled through the code.